### PR TITLE
[#13109]-IndexOutOfBoundsException encountered by LogsProcessor 

### LINKED ISSUE #13109: IndexOutOfBoundsException encountered by LogsProcessor
V9-0-0-beta.6

Found in production server logs. Happened at least 3 times so far within last 24 hours
```
Error Message: IndexOutOfBoundsException caught by WebApiServlet: Index -1 out of bounds for length 0
caused by java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 0
at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
at java.util.Objects.checkIndex(Objects.java:372)
at java.util.ArrayList.get(ArrayList.java:459)
at teammates.logic.api.LogsProcessor.getNumberOfLogsForEvent(LogsProcessor.java:88)
at teammates.ui.webapi.CalculateUsageStatisticsAction.execute(CalculateUsageStatisticsAction.java:30)
at teammates.ui.webapi.CalculateUsageStatisticsAction.execute(CalculateUsageStatisticsAction.java:17)
at teammates.ui.servlets.WebApiServlet.executeWithTransaction(WebApiServlet.java:128)
at teammates.ui.servlets.WebApiServlet.invokeServlet(WebApiServlet.java:66)
at teammates.ui.servlets.WebApiServlet.doGet(WebApiServlet.java:38)
at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:500)
at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587)
at ...
```

### PR DESCRIPTION
**Pull request by Ashwinn11**
Fixes  #13109 
- Added a check to ensure the `logs` list is not empty before accessing its last element.

### CODE REVIEW

File: src/main/java/teammates/logic/api/LogsProcessor.java [hunk 1]
```diff
@@ -84,8 +84,8 @@ public int getNumberOfLogsForEvent(Instant startTime, Instant endTime, LogEvent
             QueryLogsResults logFetchResults = queryLogs(logsParams);
             List<GeneralLogEntry> logs = logFetchResults.getLogEntries();
             total += logs.size();
-            if (logFetchResults.getHasNextPage()) {
-                logStartTime = logs.get(logs.size() - 1).getTimestamp();
+            if (logFetchResults.getHasNextPage() && !logs.isEmpty()) {
+                logStartTime = logs.getLast().getTimestamp();
```
Comment (Andy-W-Developer) [inline src/main/java/teammates/logic/api/LogsProcessor.java:88]: getLast() does not exist for Lists in JDK 11

it was added in JDK 21 https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/List.html#getLast()

Comment (Ashwinn11) [reply src/main/java/teammates/logic/api/LogsProcessor.java]: switched back to previous method

File: src/main/java/teammates/logic/api/LogsProcessor.java [hunk 2]
```diff
@@ -85,7 +85,7 @@ public int getNumberOfLogsForEvent(Instant startTime, Instant endTime, LogEvent
             List<GeneralLogEntry> logs = logFetchResults.getLogEntries();
             total += logs.size();
             if (logFetchResults.getHasNextPage() && !logs.isEmpty()) {
-                logStartTime = logs.getLast().getTimestamp();
+                logStartTime = logs.get(logs.size()-1).getTimestamp();
```
Comment (Andy-W-Developer) [inline src/main/java/teammates/logic/api/LogsProcessor.java:88]: checkstyle wants a space around the "-"

size() - 1

Comment (Ashwinn11) [reply src/main/java/teammates/logic/api/LogsProcessor.java]: fixed it.

**Review by domoberzin — APPROVED** [review]
LGTM, failing tests are most likely not caused by this change, thanks for your contribution

**Review by dishenggg — APPROVED** [review]
LGTM!

**Review by wkurniawan07 — APPROVED** [review]

**Pull request merged**